crates/autopilot/src/run_loop.rs

```diff
@@ -188,7 +192,13 @@ impl RunLoop {
         results
             .into_iter()
             .filter_map(|(index, result)| match result {
-                Ok(result) => Some((index, result)),
+                Ok(result) if result.score.is_finite() && result.score >= 0. => {
+                    Some((index, result))
+                }
+                Ok(result) => {
+                    tracing::warn!("bad score {:?}", result.score);
```

Comment:
Should this maybe also include the solver that produced the result or some other data that's helpful for following up on this?

Reply:
Probably yes but that's relevant for the other error branch too, which doesn't have it either. Something for a future PR which needs to think more about how solvers are supposed to be identified in log messages.